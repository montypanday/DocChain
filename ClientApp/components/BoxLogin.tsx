import * as React from "react";
const s = require("../css/singleLogin.css");

interface LoginState {
    loggedIn: boolean;
}

export class BoxLogin extends React.Component<{}, LoginState> {
    constructor(props) {
        super(props);
        this.state = {
            loggedIn: false
        };
    }

    public render() {

        if (location.search != "") {
            const search = location.search;
            const params = new URLSearchParams(search);
            const AuthorizationCode = params.get("code");

            fetch("/api/Login/get?authCode=" + AuthorizationCode)
                .then(response => response.json())
                .then(data => {
                    sessionStorage.setItem("OAuthSession", JSON.stringify(data));
                    sessionStorage.setItem("box_access_token", data.access_token);
                    sessionStorage.setItem("box_refresh_token", data.refresh_token);
                });
        }

        if (sessionStorage.getItem("accessToken") == null) {

            return <div className="body">
                    <div className="header">
                        <p>Click to login to Box.com</p>
                    </div>

                    <a className="logo-link" href="https://account.box.com/api/oauth2/authorize?response_type=code&client_id=3syx1zpgoraznjex526u78ozutwvgeby&state=security_token%3DKnhMJatFipTAnM0nHlZA">
                        <div className="box">
                            <section className="box-logo"></section>
                        </div>
                    </a>
                </div>;
        } else {
            return null;
        }
    }

}
