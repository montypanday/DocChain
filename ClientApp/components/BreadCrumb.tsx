﻿import * as React from "react";
import { render } from "react-dom";

export interface BreadCrumbProps {
    pathCollection: any;
    navigateOutHandler: any;
}


export class BreadCrumb extends React.Component<BreadCrumbProps, {}> {
    constructor(props) {
        super(props);

    }
    shouldComponentUpdate(nextProps) {
        const a = this.props.pathCollection !== nextProps.pathCollection;
        return a;
    }
    render() {
        var pathElements = this.props.pathCollection.map(function (row) {
            return (<a key={row.fileId}  onClick={() => { this.props.navigateOutHandler(row); }}>{row.Name}</a>);
        }.bind(this));
        return (
            <div className="breadcrumb flat">
                {pathElements}
            </div>
        );
    }
}