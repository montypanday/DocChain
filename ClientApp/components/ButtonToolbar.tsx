import * as React from "react";
import { render } from "react-dom";
import { ButtonToolbar, Button, DropdownButton, MenuItem } from "react-bootstrap";

export interface ButtonToolbarProps {
    NewFolderHandler: any;
    uploadHandler: any;
    logoutHandler: any;
}

export interface ButtonToolbarState {
}

export class ButtonToolBar extends React.Component<ButtonToolbarProps, ButtonToolbarState> {
    constructor(props: ButtonToolbarProps) {
        super(props);
    }
    shouldComponentUpdate(nextProps) {
        return false;
    }

    render() {
        return (

            <ButtonToolbar>
                <div className="btn-group">
                    <button type="button" className="btn btn-default btn-line dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                        <i style={{ paddingRight: "4px" }} className="fa fa-upload"></i>
                        Upload <span className="caret"></span>
                    </button>
                    <ul className="dropdown-menu">
                        <li>
                            <a className="btn-file">
                                <i className="fa fa-arrow-circle-o-up dropDownIcon" aria-hidden="true"></i>Upload file <input id="uploadINPUT" multiple type="file" onChange={e => { this.props.uploadHandler(e.target.files); }} />
                            </a>
                        </li>
                    </ul>
                </div>
                <div className="btn-group">
                    <button type="button" className="btn btn-default btn-line dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                        <i className="fa fa-plus dropDownIcon" aria-hidden="true"></i>New <span className="caret"></span>
                    </button>
                    <ul className="dropdown-menu">
                        <li><a onClick={this.props.NewFolderHandler} ><i className="fa fa-folder dropDownIcon" aria-hidden="true"></i>New Folder</a></li>
                    </ul>
                </div>
                <Button onClick={this.props.logoutHandler} bsStyle="danger">
                    <i className="fa fa-power-off power-icon" aria-hidden="true"></i>
                    Logout
                </Button>
            </ButtonToolbar>

        );
    }
}