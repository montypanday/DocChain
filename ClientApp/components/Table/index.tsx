import ButtonToolBar from "../FeatureBar/ButtonToolbar";
import { Row } from "./Row";
import TableHeading from "./TableHeading";

export { ButtonToolBar, Row, TableHeading };