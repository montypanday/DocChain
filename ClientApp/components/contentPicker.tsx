import * as React from 'react';
import { render } from 'react-dom';
import { ContentPicker } from 'box-ui-elements';
import messages from 'box-ui-elements/lib/i18n/en-US';
import 'box-ui-elements/dist/picker.css';

const token = 'ACCESS_TOKEN';

const getLocalizedMessage = (id, replacements) =>
    messages[id].replace(/{\s*(\w+)\s*}/g, (match, key) => replacements[key]);

interface pickerinterface { }

export class contentPicker extends React.Component<{}, pickerinterface> {
    constructor() {
        super();
        this.state = { currentCount: 0 };
    }


    public render() {
        return (
            <div>
                <ContentPicker token={token} getLocalizedMessage={getLocalizedMessage} /> , document.querySelector('.container')
        </div>
        )
    }
}
