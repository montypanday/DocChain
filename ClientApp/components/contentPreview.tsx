import * as React from 'react';
import { render } from 'react-dom';
import { ContentPreview } from 'box-ui-elements';

const token = 'ACCESS_TOKEN';
const fileId = 'FILE_ID';

interface previewinterface { }

export class contentPreview extends React.Component<{}, previewinterface> {
    constructor() {
        super();
        this.state = { currentCount: 0 };
    }
    public render() {
        return (
            <div>
                <ContentPreview fileId={fileId} token={token} />,
    document.querySelector('.container')
                </div>)
    }
}
