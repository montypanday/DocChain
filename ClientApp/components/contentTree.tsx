import * as React from 'react';
import { render } from 'react-dom';
import { ContentTree } from 'box-ui-elements';
import messages from 'box-ui-elements/lib/i18n/en-US';
import 'box-ui-elements/dist/tree.css';

const token = 'ACCESS_TOKEN';
const getLocalizedMessage = (id, replacements) =>
    messages[id].replace(/{\s*(\w+)\s*}/g, (match, key) => replacements[key]);

interface treeinterface { }

export class contentTree extends React.Component<{}, treeinterface> {
    constructor() {
        super();
        this.state = { currentCount: 0 };
    }
    public render() {
        return (
            <div>
                <ContentTree token={token}getLocalizedMessage={getLocalizedMessage}/>,document.querySelector('.container')
            </div>)
    }
}
