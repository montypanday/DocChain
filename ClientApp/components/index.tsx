import { LoadingGif } from "./loadingGif";
import { SearchBar } from "./SearchBar";
import { BreadCrumb } from "./breadCrumb";
import { BoxLogin } from "./BoxLogin";

export { LoadingGif, SearchBar, BreadCrumb, BoxLogin };
