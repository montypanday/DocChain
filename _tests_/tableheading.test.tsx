import * as React from "react";
import { TableHeading } from "../ClientApp/components/Table";
import * as renderer from "react-test-renderer";

test("Link changes the class when hovered", () => {

    const component = renderer.create(
        <TableHeading></TableHeading>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});