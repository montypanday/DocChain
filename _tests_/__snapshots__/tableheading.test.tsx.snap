// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<thead>
  <tr>
    <th
      className="col-xs-6"
    >
      File Name
    </th>
    <th
      className="col-xs-1 action-head"
    >
      Actions
    </th>
    <th
      className="col-xs-1 secure-head"
    >
      Status
    </th>
    <th
      className="col-xs-1"
    >
      Size 
    </th>
    <th
      className="col-xs-2"
    >
      Last Modified
    </th>
  </tr>
</thead>
`;
